tests/beaker_tests/netdev_stdlib/network_dns_provider_nondefaults.rb

```diff
@@ -58,64 +58,79 @@
-    on(agent, get_vshell_cmd('show run | i domain-list|name-server'))
-    stdout.scan(/ip domain-list \S+|ip name-server .*/).each do |cli|
-      command_config(agent, "no #{cli}", "removing #{cli}")
-    end
-  end
-end
-
-# @test_name [TestCase] Executes nondefaults testcase for network_dns Resource.
-test_name "TestCase :: #{testheader}" do
+def check_dns_warning
+  return unless operating_system == 'ios_xr'
   dns_warning = "
     *****************************************************************
-    *****************************************************************
     ***                    WARNING WARNING WARNING                ***
     ***                                                           ***
-    ***                                                           ***
-    ***                                                           ***
-    ***                                                           ***
     *** This test will remove all DNS settings from the testbed   ***
     *** running-config.                                           ***
     ***                                                           ***
     *** Please save the DNS settings before executing this test.  ***
     ***                                                           ***
-    ***                                                           ***
     *** Comment out the 'fail dns_warning' command below to       ***
     *** execute this test.                                        ***
     ***                                                           ***
-    ***                                                           ***
     ***                    WARNING WARNING WARNING                ***
-    *****************************************************************
     *****************************************************************"
   fail dns_warning if dns_warning
+end
 
-  ############
-  # Start Test
-  ############
-  step 'TestStep :: Testbed pre-test cleanup' do
-    dns_clean(agent)
+# Helper for XR testbed cleanup
+def xr_dns_clean(agent)
+  return unless operating_system == 'ios_xr'
+  # remove any existing resources that we will be testing against
+  resource_titles(agent, :domain_name, :clean)
+  ['no domain list test.com',
+   'no domain list test.net',
+   'no domain name switch1.test.com',
+   'no domain name switch2.test.com',
+   'no domain name-server 2001:4860:4860::8888',
+   'no domain name-server 8.8.8.8'].each { |cmd| command_config(agent, cmd) }
+end
+
+def dns_find_and_remove(agent)
+  return if operating_system == 'ios_xr'
+
+  # Find DNS commands affected by this test & remove them
+  cmds = test_get(agent, "i '^ip (domain-list|domain-name|name-server)'",
+                  :array)
+  return nil unless cmds
+
+  no_cmds = cmds.map { |cmd| "no #{cmd}" }.join(' ; ')
+  test_set(agent, no_cmds)
+  # return original commands as str
+  cmds.join(' ; ')
+end
+
+def dns_save(agent)
```

Comment:
So the changes look good, but did you test to make sure that entries that bleed through into the /etc/resolve.conf file really get removed?  When I worked on the minitest side of this, I found that I had to backup/restore the resolve.conf file to make sure none of the bleed through entries remained.